project/backend/database/models.py

```diff
@@ -3,11 +3,71 @@
 
 # Create your models here.
 
+
 class BasicUser(models.Model):
     user = models.OneToOneField(User, on_delete=models.CASCADE)
-    bio = models.CharField(max_length=200, default="This user has not told about herself / himself yet...")
+    bio = models.CharField(
+        max_length=200, default="This user has not told about herself / himself yet..."
+    )
     email_notification_preference = models.BooleanField(default=False)
     show_activity_preference = models.BooleanField(default=True)
 
     def __str__(self):
         return self.user.first_name + " " + self.user.last_name
+
+
+class Contributor(models.Model):
+    pass
+
+
+class Reviewer(models.Model):
+    pass
+
+
+class Theorem(models.Model):
+    theorem_id = models.IntegerField(primary_key=True)
+    theorem_title = models.CharField(max_length=100, null=False)
+    theorem_content = models.TextField(null=False)
+    publish_date = models.DateField()
+
+class SemanticTag(models.Model):
+    pass
+
+class WikiTag(models.Model):
+    pass
+
+class Annotation(models.Model):
+    pass
+
+class Node(models.Model):
+    node_id = models.IntegerField(primary_key=True)
+    node_title = models.CharField(max_length=100)
+    contributors = models.ManyToManyField(Contributor)
+    theorem = models.OneToOneField(Theorem, null=True, on_delete=models.SET_NULL)
+    publish_date = models.DateField()
+    reviewers = models.ManyToManyField(Reviewer)
+    referenced_nodes = models.ManyToManyField("self", symmetrical=True)
+    semantic_tags = models.ManyToManyField(SemanticTag)
+    wiki_tags = models.ManyToManyField(WikiTag)
+    annotations = models.ManyToManyField(Annotation)
+    is_valid = models.BooleanField()
+    num_visits = models.IntegerField()
+
```

Comment:
Proofs should be added as a one-to-many relation.
Nodes that have referenced this node should be added as a many to many relation to 'self'.

Reply:
I updated the code accordingly in the next commit.